src/app/components/core/configuration-manager/configuration-group/configuration-group.component.ts

```diff
@@ -0,0 +1,181 @@
+
+  ngOnInit() { }
+
+  ngOnChanges() {
+    this.categeryConfiguration();
+    this.getChildConfigData();
+  }
+
+  public updateCategroyConfig(config) {
+    this.category.config = config;
+    this.getChildConfigData();
+    this.categeryConfiguration();
+  }
+
+  categeryConfiguration() {
+    this.groups = [];
+    const configItems = Object.keys(this.category.config).map(k => {
+      this.category.config[k].key = k;
+      return this.category.config[k];
+    });
+
+    this.groups = chain(configItems).groupBy(x => x.group).map((v, k) => {
+      if (k != "undefined") {
+        return { category: this.category.name, group: k, config: Object.assign({}, ...v.map(vl => { return { [vl.key]: vl } })) }
+      } else {
+        return { category: this.category.name, group: "Default Configuration", config: Object.assign({}, ...v.map(vl => { return { [vl.key]: vl } })) }
+      }
+    }).value();
+  }
+
+  /**
+   * Set configuration of the selected child category
+   * @param category Object{key, description, displayName}
+   */
+  selectTab(tab: string) {
+    if (tab !== this.selectedGroup) {
+      this.selectedGroup = tab;
+    }
+  }
+
+  public getChildConfigData() {
+    // No advance configuration on add form
+    if (this.pages.includes(this.from) && this.category) {
+      this.categoryKey = this.category.name;
+      this.checkIfAdvanceConfig(this.category.name)
+    }
+  }
+
+  checkIfAdvanceConfig(categoryName: string) {
+    this.configService.getCategoryConfigChildren(categoryName).
+      subscribe(
+        (data: any) => {
+          const categoryChildren = data.categories?.filter(cat => (cat.key == `${this.categoryKey}Advanced`) || (cat.key == `${this.categoryKey}Security`));
+          categoryChildren.forEach(cat => {
+            // Set group of advance/security configuration
+            cat.group = cat?.key.includes(`${this.categoryKey}Advanced`) ? 'Advanced Configuration' :
+              (cat?.key.includes(`${this.categoryKey}Security`) ? 'Security Configuration' : cat?.displayName);
+            // Get child category configuration
+            this.getConfig(cat);
+          });
+        },
+        error => {
+          console.log('error ', error);
+        }
+      );
+  }
+
+  /**
+  * Get configuration of the child category
+  * @param categoryName : String
+  */
+  getConfig(category: any) {
+    this.configService.getCategory(category.key).
+      subscribe(
+        (data: any) => {
+          if (category.key == `${this.categoryKey}Advanced`) {
+            this.advanceConfiguration = { key: category.key, config: cloneDeep(data) };
+          }
+          if (category.key == `${this.categoryKey}Security`) {
+            this.securityConfiguration = { key: category.key, config: cloneDeep(data) };
+          }
+
+          this.upsertAdvanceConfiguration(this.groups, { category: category.key, group: category.group, config: data });
+        },
+        error => {
+          console.log('error ', error);
```

Comment:
I wonder if this should not an alert with error body?

Reply:
We can add but this is to fetch the advance or security configuration only.